src/model_theory/terms_and_formulas.lean

```diff
@@ -170,32 +211,120 @@ nonempty.intro (by constructor)
 variables (M)
 
 /-- A bounded formula can be evaluated as true or false by giving values to each free variable. -/
-@[simp] def realize_bounded_formula :
+def realize_bounded_formula :
   ∀ {l} (f : L.bounded_formula α l) (v : α → M) (xs : fin l → M), Prop
 | _ bd_falsum  v     xs := false
 | _ (bd_equal t₁ t₂) v xs := realize_term (sum.elim v xs) t₁ = realize_term (sum.elim v xs) t₂
 | _ (bd_rel R ts)   v   xs := rel_map R (λ i, realize_term (sum.elim v xs) (ts i))
 | _ (bd_imp f₁ f₂)  v xs := realize_bounded_formula f₁ v xs → realize_bounded_formula f₂ v xs
-| _ (bd_all f)     v   xs := ∀(x : M), realize_bounded_formula f v (fin.cons x xs)
+| _ (bd_all f)     v   xs := ∀(x : M), realize_bounded_formula f v (fin.snoc xs x)
+
+namespace bounded_formula
+
+variables {M} {l : ℕ} (φ ψ : L.bounded_formula α l) (θ : L.bounded_formula α l.succ)
+variables (v : α → M) (xs : fin l → M)
+
+@[simp] lemma realize_bot :
+  realize_bounded_formula M (⊥ : L.bounded_formula α l) v xs = false :=
```

Comment:
I'd reformulate all equalities in this section with `↔`. Although it doesn't matter much, that is standard in mathlib, and `↔` has a nicer precedence than `=` when dealing with propositions.

Reply:
You can also consider making all arguments implicit